lib/saluki-components/src/transforms/dogstatsd_mapper/mod.rs

```diff
@@ -0,0 +1,373 @@
+        let metric_name = context.name();
+        let tags = context.tags();
+        let origin_tags = context.origin_tags();
+
+        for profile in &self.profiles {
+            if !metric_name.starts_with(&profile.prefix) && profile.prefix != "*" {
+                continue;
+            }
+
+            for mapping in &profile.mappings {
+                if let Some(captures) = mapping.regex.captures(metric_name) {
+                    let mut name = String::new();
+                    captures.expand(&mapping.name, &mut name);
+                    let mut new_tags: Vec<String> = tags.into_iter().map(|tag| tag.as_str().to_owned()).collect();
+                    for (tag_key, tag_value_expr) in &mapping.tags {
+                        let mut expanded_value = String::new();
+                        captures.expand(tag_value_expr, &mut expanded_value);
+                        new_tags.push(format!("{}:{}", tag_key, expanded_value));
+                    }
+                    return self
+                        .context_resolver
+                        .resolve_with_origin_tags(&name, new_tags, origin_tags.clone());
+                }
+            }
+        }
+        None
+    }
+}
+
+impl DogstatsDMapperConfiguration {
+    /// Creates a new `DogstatsDMapperConfiguration` from the given configuration.
+    pub fn from_configuration(config: &GenericConfiguration) -> Result<Self, GenericError> {
+        Ok(config.as_typed()?)
+    }
+}
+
+#[async_trait]
+impl SynchronousTransformBuilder for DogstatsDMapperConfiguration {
+    async fn build(&self) -> Result<Box<dyn SynchronousTransform + Send>, GenericError> {
+        let metric_mapper = self
+            .dogstatsd_mapper_profiles
+            .build(self.context_string_interner_bytes)?;
+        Ok(Box::new(DogstatsDMapper { metric_mapper }))
+    }
+}
+
+impl MemoryBounds for DogstatsDMapperConfiguration {
+    fn specify_bounds(&self, builder: &mut MemoryBoundsBuilder) {
+        builder
+            .minimum()
+            // Capture the size of the heap allocation when the component is built.
+            .with_single_value::<DogstatsDMapper>("component struct")
+            // We also allocate the backing storage for the string interner up front, which is used by our context
+            // resolver.
+            .with_fixed_amount("string interner", self.context_string_interner_bytes.as_u64() as usize);
+    }
+}
+
+pub struct DogstatsDMapper {
+    metric_mapper: MetricMapper,
+}
+
+impl SynchronousTransform for DogstatsDMapper {
+    fn transform_buffer(&mut self, event_buffer: &mut FixedSizeEventBuffer) {
+        for event in event_buffer {
+            if let Some(metric) = event.try_as_metric_mut() {
+                if let Some(new_context) = self.metric_mapper.try_map(metric.context()) {
+                    *metric.context_mut() = new_context;
+                }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I think we're missing some simple tests to exercise the error cases:

- unknown match type
- invalid match string when using `wildcard` match type
- consecutive asterisks when using `wildcard` match type

Reply:
Done. Included all the same tests from the [datadog-agent mapper tests](https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/mapper/mapper_test.go) so the expected behavior *should* be the same